.github/workflows/R-CMD-check.yaml

```diff
@@ -43,10 +43,21 @@ jobs:
 
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          extra-packages: any::rcmdcheck
+          extra-packages: any::rcmdcheck,covr,lintr,styler
           needs: check
 
       - uses: r-lib/actions/check-r-package@v2
         with:
           upload-snapshots: true
           build_args: 'c("--no-manual","--compact-vignettes=gs+qpdf")'
+  
+      - name: Run test coverage
+        run: Rscript -e 'covr::codecov()'
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+
+      - name: Run linting
+        run: Rscript -e 'lintr::lint_package()'
+
+      - name: Check code formatting with styler
```

Comment:
is there any tooling (e.g. extensions) for R studio or VS Code to do apply the same formatting automatically? and/or is it possible to add a pre-commit hook to format on commit? E.g. for python we use a [black pre-commit hook](https://github.com/noisepy/NoisePy/blob/2b7627dbc8464ef56bf8565c95daa7b1104f2bc4/.pre-commit-config.yaml#L34) and we also use the VSCode black extension

Reply:
you can also create a task to add this in a subsequent PR

Reply:
I see you already have a draft for adding pre-commit :). It will be good to also figure out the VS Code/R studio formatting and linting support for R.

Reply:
create a new PR to address the comments in a cleaner, more organized way. Closing this PR